src/mlpack/core/util/size_checks.hpp

```diff
@@ -31,16 +31,43 @@ template<typename DataType, typename LabelsType>
 inline void CheckSameSizes(const DataType& data,
                            const LabelsType& label,
                            const std::string& callerDescription,
-                           const std::string& addInfo = "labels")
-{
-  if (data.n_cols != label.n_cols)
+                           const std::string& addInfo = "labels",
+                           const std::string& isTranspose = "none")
+{ 
+  if (isTranspose == "data")
```

Comment:
Another way to handle this might be something like this:

```
const size_t dataPoints = (isTranspose == "data") ? data.n_rows : data.n_cols;
const size_t labelPoints = (isTranspose == "labels") ? labels.n_rows : labels.n_cols;
```
and then use those values in the rest of the function.